lib/reversetunnel/resolver.go

```diff
@@ -40,21 +40,21 @@ func CachingResolver(ctx context.Context, resolver Resolver, clock clockwork.Clo
 	if err != nil {
 		return nil, err
 	}
-	return func(ctx context.Context) (*utils.NetAddr, error) {
+	return func(ctx context.Context) (utils.NetAddr, error) {
 		a, err := cache.Get(ctx, "resolver", func(ctx context.Context) (interface{}, error) {
 			return resolver(ctx)
 		})
 		if err != nil {
-			return nil, err
+			return utils.NetAddr{}, err
 		}
-		return a.(*utils.NetAddr), nil
+		return *a.(*utils.NetAddr), nil
```

Comment:
Hmm should we check for nil here actually?

Reply:
Is this correct? 
Shouldn't it be `a.(utils.NetAddr)`?

Reply:
I wondered that too, so I tested in a go sandbox and I get a runtime panic doing that since the underlying type is *T not T

https://go.dev/play/p/XsSLh1mEjjE

Reply:
But your use case is that `resolve` function returns a `utils.NetAddr` and not a `*utils.NetAddr`
This is similar to what you have: https://go.dev/play/p/CC-3PDS-M5J